### PR TITLE
Add logic when the show bumper.

### PR DESCRIPTION
**Pull request by oksana-slu**

### CODE REVIEW

**Comment by auraz** [comment]
@polesye please review

File: common/lib/xmodule/xmodule/js/src/video/09_bumper.js [hunk 1]
```diff
@@ -79,6 +81,7 @@ define(
             var events = ['ended', 'skip', 'error'].join(' ');
             this.element.off(events, this.showMainVideo);
             if (_.isFunction(this.state.videoPlayer.destroy)) {
+                this.state.videoSaveStatePlugin.onSkip(this.doNotShowAgain);
```
Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_bumper.js]: Can we use `this.state.el.trigger('skip', [this.doNotShowAgain])`?

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js [hunk 2]
```diff
@@ -91,6 +93,14 @@ function() {
             this.saveState(true, {youtube_is_available: youtubeIsAvailable});
         },
 
+        onSkip: function (doNotShowAgain) {
+            if (doNotShowAgain) {
+                this.saveState(true, {do_not_show_again_bumper: true});
+            } else {
+                this.saveState(true, {date_last_view_bumper: true});
```
Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: I think we need `date_last_view_bumper ` for both requests?

**Comment by polesye** [comment]
Does video position work well with these changes?

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js (hunk 2 above)
Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]:
```javascript
var info = {data_last_view_bumper: true};
if (doNotShowAgain) {
    _.extend(info, {do_not_show_again: true});
}
this.saveState(true, info);
```
Is it better?

File: common/lib/xmodule/xmodule/js/src/video/09_bumper.js (hunk 1 above)
Comment (oksana-slu) [inline common/lib/xmodule/xmodule/js/src/video/09_bumper.js]: done

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js (hunk 2 above)
Comment (oksana-slu) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: done

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js [hunk 3]
```diff
@@ -91,8 +93,19 @@ function() {
             this.saveState(true, {youtube_is_available: youtubeIsAvailable});
         },
 
+        onSkip: function (event, doNotShowAgain) {
+            var info = {date_last_view_bumper: true};
+            if (doNotShowAgain) {
+                _.extend(info, {do_not_show_again_bumper: true});
+            }
+            this.saveState(true, info);
```
Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: @auraz do you agree that we should send `date_last_view_bumper ` in both cases?

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js [hunk 4]
```diff
@@ -91,8 +93,19 @@ function() {
...
+        },
+
         saveState: function (async, data) {
             if (!($.isPlainObject(data))) {
+                if (this.state.isBumper) {
```
Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: @auraz Do we still need to send info on play/pause and other events?

Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: @oksana-slu In this case, it does not save the state onSkip. Am I right?

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js (hunk 3 above)
Comment (auraz) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: @polesye  I will not be error, but why we need to update last viewed time on backend if we will turn it off?

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js (hunk 4 above)
Comment (auraz) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: @polesye yes, because it is done in destroy anyway

Comment (oksana-slu) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: fix it "Does video position work well with these changes?"

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js (hunk 3 above)
Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: I think, we should send this data when we resolve the videoBumper ('skip', 'ended', 'error' events).

File: common/lib/xmodule/xmodule/js/src/video/09_bumper.js [hunk 5]
```diff
@@ -79,6 +81,7 @@ define(
             var events = ['ended', 'skip', 'error'].join(' ');
             this.element.off(events, this.showMainVideo);
             if (_.isFunction(this.state.videoPlayer.destroy)) {
+                this.state.el.trigger('skip', [this.doNotShowAgain]);
```
Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_bumper.js]: I think `skip` is incorrect event here.

Comment (polesye) [inline common/lib/xmodule/xmodule/js/src/video/09_bumper.js]: you can remove this line.

File: common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js (hunk 3 above)
Comment (auraz) [inline common/lib/xmodule/xmodule/js/src/video/09_save_state_plugin.js]: @polesye why not on destroy at once?

File: common/lib/xmodule/xmodule/js/src/video/09_bumper.js (hunk 1 above)
Comment (auraz) [inline common/lib/xmodule/xmodule/js/src/video/09_bumper.js]: @polesye  why? This is related to the question about place we should fire saveSate.

Comment (oksana-slu) [inline common/lib/xmodule/xmodule/js/src/video/09_bumper.js]: я пробывала вызывать этот метод (this.state.videoSaveStatePlugin.onSkip(this.doNotShowAgain);) отдельно в функциях skip и  skipAndDoNotShowAgain, но была ошибка, что videoSaveStatePlugin не существует, поэтому оставила  в destroy

**Pull request merged**